File: f1-predictor/frontend/src/utils/teamColors.ts
export const TEAM_COLORS: Record<string, string> = {
  mclaren: '#FF8000',
  red_bull: '#3671C6',
  ferrari: '#E8002D',
  mercedes: '#27F4D2',
  aston_martin: '#229971',
  alpine: '#FF87BC',
  haas: '#B6BABD',
  rb: '#6692FF',
  williams: '#64C4FF',
  sauber: '#52E252',
  cadillac: '#C9CDD1',
};

export const getTeamColor = (teamId: string): string =>
  TEAM_COLORS[teamId.toLowerCase()] ?? '#9CA3AF';

File: f1-predictor/backend/app/utils/team_colors.py
"""2025 F1 team colour mapping for UI display."""

TEAM_COLORS: dict[str, str] = {
    "mclaren": "#FF8000",
    "ferrari": "#E8002D",
    "red_bull": "#3671C6",
    "mercedes": "#27F4D2",
    "aston_martin": "#229971",
    "alpine": "#FF87BC",
    "williams": "#64C4FF",
    "rb": "#6692FF",
    "sauber": "#52E252",
    "haas": "#B6BABD",
    "cadillac": "#004785",
}


def team_color(team_id: str, team_name: str | None = None) -> str:
    key = team_id.lower().replace("-", "_")
    if key in TEAM_COLORS:
        return TEAM_COLORS[key]
    if team_name:
        name_key = team_name.lower().replace(" ", "_")
        for token in (name_key, name_key.split("_")[0]):
            if token in TEAM_COLORS:
                return TEAM_COLORS[token]
    return "#888888"
